### PR TITLE
NF: Add deep clone to deck config

### PR DESCRIPTION
**Pull request by Arthur-Milchior**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java [hunk 1]
```diff
@@ -61,4 +66,11 @@ public boolean isDyn() {
     public boolean isStd() {
         return getInt("dyn") == Consts.DECK_STD;
     }
+
+
+    @Override
+    public JSONObject deepClone() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java:72]: Any way we can return `DeckConfig` here?

File: AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java [hunk 2]
```diff
@@ -22,6 +22,11 @@
 import timber.log.Timber;
 
 public class DeckConfig extends JSONObject{
+
+    public DeckConfig() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java:26]: Is this necessary? Thought an empty `ctor` is generated by the compiler

File: AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java]: That was indeed the point. Bad copy paste

File: AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java:26]: It's only the case if there is no other constructors. In these cases there are other constructors

Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java:26]: Do we want this to be public?

**Review by david-allison-1 — APPROVED** [review]
Either way, LGTM

**Review by mikehardy — APPROVED** [review]
LGTM

**Pull request merged**